google/cloud/options.h

```diff
@@ -306,6 +307,40 @@ void CheckExpectedOptions(Options const& opts, char const* caller) {
  */
 Options MergeOptions(Options a, Options b);
 
+/**
+ * The prevailing options for the current operation.
+ */
+Options const& CurrentOptions();
+
+/**
+ * RAII object to set/restore the prevailing options for the enclosing scope.
+ *
+ * @code
+ * struct IntOption { using Type = int; };
+ * assert(!internal::CurrentOptions().has<IntOption>());
+ * {
+ *   internal::OptionsSpan span(Options{}.set<IntOption>(1));
+ *   assert(internal::CurrentOptions().get<IntOption>() == 1);
+ *   {
+ *     internal::OptionsSpan span(Options{}.set<IntOption>(2));
+ *     assert(internal::CurrentOptions().get<IntOption>() == 2);
+ *   }
+ *   assert(internal::CurrentOptions().get<IntOption>() == 1);
+ * }
+ * assert(!internal::CurrentOptions().has<IntOption>());
+ * @endcode
+ *
+ * @param opts the `Options` to install.
+ */
+class ABSL_MUST_USE_RESULT OptionsSpan {
```

Comment:
Q: Is `ABSL_MUST_USE_RESULT` needed here, or nice to have, or... what's this here for?

Reply:
It is there to try to call out usages like `OptionsSpan(opts);` where you forget the variable name, and the temporary span dies at the semicolon.  For example, see https://github.com/abseil/abseil-cpp/blob/master/absl/base/attributes.h#L410-L414.  (Previously we might have used a macro trick to do something similar.)

That said, when I tried to test that, it didn't seem to trigger a warning.  But I'm not sure that wasn't just an artifact of which compiler+version I used, so I decided to leave it anyway.  But, happy to remove also.